src/solver/reduce/forward_all.cpp

```diff
@@ -0,0 +1,276 @@
+
+        result.construction_params.push_back(kernel);
+    }
+
+    {
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(output_numel, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        auto kernel = KernelInfo{};
+
+        kernel.kernel_file = "MIOpenReduceCalculation.cpp";
+        kernel.kernel_name = "CalculationFwdContiguous";
+
+        const auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_INT8", static_cast<int32_t>(dtype == miopenInt8)},
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+            {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"OUTPUT_TYPE", input_dtype == "uint8_t" ? "uint8_t" : "bool"},
+            {"OP_TYPE", "ReduceCalculationOp_t::lAND"},
+            {"CALCULATION_DTYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"IS_LOGICAL_CALCULATION", "true"},
+            {"IS_USE_FLOAT_DTYPE", problem.IsValidFloatTypes()},
+            {"MIOPEN_REDUCE_CALCULATION_PROD", MIOPEN_REDUCE_CALCULATION_PROD},
+            {"MIOPEN_REDUCE_CALCULATION_SUM", MIOPEN_REDUCE_CALCULATION_SUM},
+            {"MIOPEN_REDUCE_CALCULATION_ANY", MIOPEN_REDUCE_CALCULATION_ANY},
+            {"MIOPEN_REDUCE_CALCULATION_ALL", MIOPEN_REDUCE_CALCULATION_ALL}};
+
+        kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+        kernel.l_wk.push_back(xlocalsize);
+        kernel.l_wk.push_back(ylocalsize);
+        kernel.l_wk.push_back(zlocalsize);
+
+        kernel.g_wk.push_back(xgridsize);
+        kernel.g_wk.push_back(ygridsize);
+        kernel.g_wk.push_back(zgridsize);
+
+        result.construction_params.push_back(kernel);
+    }
+
+    if(is_parallelism(reqd_work_item_cnt, output_numel, reduce_size))
+    {
+        result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+                decltype(auto) parallel_kernel = handle_.Run(kernels[0]);
+                decltype(auto) kernel          = handle_.Run(kernels[1]);
+                decltype(auto) params =
+                    raw_params.CastTo<miopen::reduce::CalculationInvokeParams>();
+
+                auto xdims = params.xDesc->GetLengths();
+                auto ydims = params.yDesc->GetLengths();
+                auto dim   = params.dim;
+
+                auto reduce_size = xdims[dim];
+                auto output_numel =
+                    std::accumulate(ydims.begin(), ydims.end(), 1ULL, std::multiplies<size_t>());
+
+                auto inner_size = std::accumulate(
+                    xdims.begin() + dim + 1, xdims.end(), 1ULL, std::multiplies<size_t>());
```

Comment:
ditto

Reply:
solved.

notice that, for inner_size, check my comment above, we cannot use GetElementSize() because inner_size accumulate from dim+1 to end